docs/guide/tutorial-yii-integration.md

```diff
@@ -114,6 +118,24 @@ Like in a Yii application, you should configure the application instance based o
 the third-party system. For example, to use the [Active Record](db-active-record.md) feature, you need to configure
 the `db` [application component](structure-application-components.md) with the DB connection setting used by the third-party system.
 
+While registering the components (assume that we are using Yii in other non-Yii app which does not use [yii2-composer](https://github.com/yiisoft/yii2-composer)) return the object of that composnent instead of array or annonymous function as explained in [guide](https://www.yiiframework.com/doc/guide/2.0/en/structure-application-components). Example
+
+```php
+// web.php yii config file
+'components' => [
+    'radar' => require __DIR__.'/../../app/My/MyClass.php'
+    
+    
+// MyClass.php
+namespace App\Custom;
+
+use \some\class\Here;
+
+return new MyClass  extends \yii\base\BaseObject // <-- annonymous class is supported in PHP >= 7
```

Comment:
That looks interesting. What's the purpose of doing that instead of using regular class names?

Reply:
https://github.com/yiisoft/yii2/pull/17795#issuecomment-573321975